xbmc/filesystem/CurlFile.cpp

```diff
@@ -565,6 +560,19 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
       g_curlInterface.easy_setopt(h, CURLOPT_HTTPAUTH, CURLAUTH_DIGEST);
     else if( m_httpauth == "ntlm" )
       g_curlInterface.easy_setopt(h, CURLOPT_HTTPAUTH, CURLAUTH_NTLM);
+    else
+      bAuthSet = false;
+  }
+
+  // set username and password for current handle
+  if (!m_username.empty())
+  {
+    g_curlInterface.easy_setopt(h, CURLOPT_USERNAME, m_username.c_str());
+    if (!m_password.empty())
+      g_curlInterface.easy_setopt(h, CURLOPT_PASSWORD, m_password.c_str());
```

Comment:
From looking at the docs when using these newer opts which separate username and password the values will be URL decoded. Therefore I think you should URL encode the values beforehand.

Ref: https://curl.se/libcurl/c/CURLOPT_USERPWD.html

Reply:
Are you sure? I read the docs differently.

Reply:
```
The user and password strings are not URL decoded, so there's no way to send in a user name containing a colon using this option. Use CURLOPT_USERNAME for that, or include it in the URL.
```

Reply:
I read that as only relevant for the old style. There's no mention of encoding on the 2 new style pages at all.
As the new style is 2 separate calls (use and pass) there's no need for a distinction for the colon separated like the old style

Reply:
I had a quick look at the curl sources and found no trace of user/pass strings being decoded.

Reply:
Ok, fuzzard’s explanation makes sense. Sorry for the noise.